upload/validation.go

```diff
@@ -21,6 +21,11 @@ func getServiceDescriptor(contentType string) (*validators.ServiceDescriptor, er
 			Service:  "advisor",
 			Category: "upload",
 		}, nil
+	case "application/gzip; charset=binary":
```

Comment:
All these cases can be consolidated into a single statement separated by commas. This way you aren't repeating the return statement.

```go
switch ctype := contentType; ctype {
    case "application/x-gzip; charset=binary", "application/gzip", "application/gzip; charset=binary":
        return &validators.ServciceDescription{Service: "advisor", Category: "upload"}, nil
    default:
        /* ... */
```

ref: https://golang.org/ref/spec#Switch_statements

Reply:
Another option could be to change the switch to something like this, though if you anticipate there being more fixed `ServiceDescription` types in the future, the switch is more future-proof.

```go
if strings.HasPrefix("application/gzip", contentType) || strings.HasPrefix("application/x-gzip", contentType) {
    return &validators.ServiceDescription{Service: "advisor", Category: "upload"}, nil
} else {
    /* ... */
}
```